### ISSUE TITLE
Accessors After Function Calls Fail With No Error

### ISSUE THREAD

**Issue by tinkerer-red** [YoYoGames/GameMaker-Bugs#7799]
### Description

Using a function call to fetch a data structure, then immediately accessing that structure throws an error.

The only exception to this would be the dot accessor - this will fail to compile with no error message.

### Expected Change

Accessors should allow for function calls as input, or the ide should throw an error if this is an expected result

### Steps To Reproduce

```gml
// all functions listed would simply be getter functions
func()[arrIndex()]
func()[@ arrIndex()]
func()[| listIndex()]
func()[? mapKey()]
func()[$ structKey()]
func()[# gridX(), gridY()]
```

Alternative to get this to work;
```gml
var _t = func(); _t[arrIndex()]
var _t = func(); _t[@ arrIndex()]
var _t = func(); _t[| listIndex()]
var _t = func(); _t[? mapKey()]
var _t = func(); _t[$ structKey()]
var _t = func(); _t[# gridX(), gridY()]
```

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.8.1 (Monthly), 2024.1100 (Betas)

### Which platform(s) are you seeing the problem on?

Windows

### Sample Package Attached?

- [X] I have attached my bug package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by rwkay** [comment]
I cannot find an issue here... this works in small examples that I have tried (based on your examples).

Please include a project that shows the problem that you are seeing.

**Comment by scott-dunbar** [comment]
This issue has been resolved but it has spawned issue https://github.com/YoYoGames/GameMaker-Bugs/issues/7998 in its wake

Closing this issue as the specifics of it have been addressed in  IDE v2024.1100.0.646  Runtime v2024.1100.0.666
